app/main.py

```diff
@@ -1 +1,58 @@
-#  write your code here
+class Animal:
+
+    def __init__(self,
+                 name: str,
+                 appetite: int,
+                 is_hungry: bool = True
+                 ) -> None:
+        self.name = name
+        self.appetite = appetite
+        self.is_hungry = is_hungry
+
+    def print_name(self) -> None:
+        print(f"Hello, I'm {self.name}")
+
+    def feed(self) -> int:
+        if self.is_hungry:
+            print(f"Eating {self.appetite} food points...")
+            self.is_hungry = False
+            return self.appetite
+
+        return 0
+
+
+class Cat(Animal):
+
+    def __init__(self,
+                 name: str,
+                 is_hungry: bool = True
+                 ) -> None:
+        super().__init__(name, 3, is_hungry)
+
+    @staticmethod
+    def catch_mouse() -> None:
+        print("The hunt began!")
+
+
+class Dog(Animal):
+
+    def __init__(self,
+                 name: str,
+                 is_hungry: bool = True
+                 ) -> None:
+        super().__init__(name, 7, is_hungry)
+
+    @staticmethod
+    def bring_slippers() -> None:
+        print("The slippers delivered!")
+
+
+def feed_animals(animals: list[Animal]) -> int:
+    feed_points_sum = 0
+
+    for animal in [animal for animal in animals
+                   if animal.is_hungry]:
```

Comment:
You don't need to make additional check `if animal.is_hungry`, because you make it inside `feed`. Also, you can rewrite this function to one line with using function `sum`